### DISCUSSION TITLE
HTTPEndpoint based workflows cluster

### DISCUSSION THREAD

**Discussion by maxche91** [elsa-workflows/elsa-core#4254] [Q&A] [answered]
Hello, I'm sorry, I'm a noob in clustering , I have workflows that start with http endpoints and I'd like to host elsa-core app in a multi-node environment for failover on my own servers. Could someone describe in a nutshell if possible, how can I arrange it? Do I need any third party technology? Unfortunately, I could not figure out how to do this from the available documentation.

**Comment by sfmskywalker** [comment, marked as answer]
Elsa itself does not provide any clustering, but it supports being hosted in a multinode environment, such as Kubernetes or Azure Container Services.

If you know how to setup a cluster that hosts ASP.NET Core apps, you shouldn't have to do anything else in order to expose workflows via HTTP Endpoint activities. After all, these workflows are invoked through ASP.NET Core middleware, which is processed just like e.g. MVC controller middleware.

However, when using Elsa 2, which uses caching & distributed cache eviction, you will also have to setup some additional infrastructure to allow application instances to communicate with one another. Details can be found here: https://v2.elsaworkflows.io/docs/hosting/distributed-hosting

**Reply by maxche91** [reply]
Thank you very much for so quick response! Could please tell if I can use Kafka as a message broker instead of RabbitMQ?

**Reply by sfmskywalker** [reply]
I don’t have any hands on experience doing it with Kafka, but since Elsa 2 uses Rebus to abstract away underlying message brokers, which in turn supports Kafka, it should be possible.

**Reply by maxche91** [reply]
Thank you!!!

**Comment by maxche91** [comment]
The last question please, sorry if I asking something stupid, but you said that I need these additional configurations/actions (service broker bus, distributed lock…) regarding Elsa 2 and yes I using 2 version, but in case if I’ll move to Elsa 3, will it work out from the box?

**Reply by sfmskywalker** [reply]
Elsa 3 is likely to also need some configuration when hosting in a cluster. Although distributed locking isn’t used as extensively, it is still used for certain scenarios. Which means you’ll need to configure a distributed lock provider such as SQL Server or Redis. As for service bus, this is not yet currently used, but it might be added later if it turns out we need something to send signals across nodes. And as for Quartz.NET, the same instructions apply as for Elsa 2.

**Reply by maxche91** [reply]
Thanks a million!
